### PR TITLE
Cherry-pick changes from main and hotfix to main-preview from Dec 9, 2024 to Dec 17, 2024

### PR DESCRIPTION
**Pull request by manvkaur**
cherry-pick following changes from main to preview from Dec 9, 2024 to Dec 11, 2024
- b561934 pin DurableTask.SqlServer to 1.4.0 (#460)                                                                                                    
- 9874ae8 remove dead code and update packages (#459)
- ef2a37f Excluding unlisted NuGet pacakges from version evaluation (#452)
- b7128c5 codeql: remove dead code to resolve code ql issue (#456)
- d995cd1 Update build status (#457)
- 30693a7 (tag: 4.26.2-Preview, origin/release/v4.x-preview-hotfix) Extension bundle HotFix 4.26.2 (#451)
- Update CodeOwners
- Update pipelines to include main branches in the triggers
- Removed unused pipeline file in base directory

### CODE REVIEW

**Comment by gavin-aguiar** [comment]
Can you also update the CODEOWNERS file?

**Review by gavin-aguiar — APPROVED** [review]

File: eng/official-build.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@ trigger:
   branches:
     include:
     - main
+    - main-preview
+    - main-v2
+    - main-v3
     - release/v4.x
     - release/v3.x
     - release/v2.x
```
Comment (manvkaur) [inline eng/official-build.yml:10]: @castrodd , I am keeping the release pipelines in triggers for now, we will need to update/remove these when we are ready to completely clean them up.

Comment (castrodd) [reply eng/official-build.yml:10]: Sounds good. Thank you!

**Review by satvu — APPROVED** [review]

**Review by jviau — APPROVED** [review]
Signing off on the codeowners change only. I did not review the rest of the PR.

**Pull request merged**